### PR TITLE
Add dist all style and refactor table create

### PR DESCRIPTION
**Pull request by ChrisPRobinson**
1) Created a Redshift table type
2) Moved code that does sql creation into Redshift table
3) Updated Redshift output to use redshift table to create tables
4) Implemented a create table with various capabilities (backup, like,
distkey, sortkeys, and diststyle)

### CODE REVIEW

File: lib/etl/redshift/table.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        type_ary = []
+        columns.each do |name, column|
+          column_type = col_type_str(column)
+          column_statement = "\"#{name}\" #{column_type}"
+          column_statement += " NOT NULL" if @primary_key.include?(name.to_sym)
+
+          type_ary << column_statement
+        end
+
+        if @primary_key.length > 0
```
Comment (emya) [inline lib/etl/redshift/table.rb]: Where do we set primary keys?

File: lib/etl/redshift/table.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+  module Redshift
+
+    # Represents single data table in redshift
+    class Table < ETL::Schema::Table
+      attr_accessor :backup, :dist_key, :sort_keys, :dist_style
+
+      def initialize(name="", opts = {})
+        super(name, opts)
+        @dist_key = ""
```
Comment (emya) [inline lib/etl/redshift/table.rb]: I think we should use symbol instead of string.

Comment (emya) [inline lib/etl/redshift/table.rb]: I saw symbols are used for primary/sort keys in the test, but dist key is string. I want to keep consistency.

Comment (ChrisPRobinson) [inline lib/etl/redshift/table.rb]: Fixed

**Pull request merged**